pkg/parse/annotations.go

```diff
@@ -33,18 +33,37 @@ type sourceContext struct {
 	Rev    string `json:"rev,omitempty"`
 }
 
+const (
+	skipOwningInventoryAnnotations = "configsync.gke.io/skip-owning-inventory-annotations"
+)
+
 func addAnnotationsAndLabels(objs []ast.FileObject, scope declared.Scope, syncName string, sc sourceContext, commitHash string) error {
 	gcVal, err := json.Marshal(sc)
 	if err != nil {
 		return fmt.Errorf("marshaling sourceContext: %w", err)
 	}
+
 	var inventoryID string
 	if scope == declared.RootReconciler {
 		inventoryID = applier.InventoryID(syncName, configmanagement.ControllerNamespace)
 	} else {
 		inventoryID = applier.InventoryID(syncName, string(scope))
 	}
 	for _, obj := range objs {
+		if string(core.GetAnnotation(obj, skipOwningInventoryAnnotations)) == "true" {
+			// Don't expose this annotation on the applied object
+			core.RemoveAnnotations(obj, skipOwningInventoryAnnotations)
+
+			// FIXME: Setting labels to an empty map to make tests pass
+			//        This is required because the normalization that happens on fake objects at pkg/testing/fake/build.go:83
+			if obj.GetLabels() == nil {
+				obj.SetLabels(make(map[string]string))
+			}
```

Comment:
I'm not super happy about this, but don't know how to avoid it. Basically, the test code [runs this on all fake objects](https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/pkg/testing/fake/build.go#L83-L85), so if we don't do it on the real ones, the diff tests fail.

I tried writing a transformer that would remove keys from maps if their values were empty maps, but couldn't get it to work properly; perhaps that is something to explore as an improvement to the test suite in general (instead of implementing for a one-off for this specific test).